arangod/Cluster/ServerState.cpp

```diff
@@ -626,8 +596,26 @@ bool ServerState::registerAtAgencyPhase1(AgencyComm& comm, ServerState::RoleEnum
   VPackSlice servers = result.slice()[0].get(
       std::vector<std::string>({AgencyCommManager::path(), "Plan", agencyListKey}));
   if (!servers.isObject()) {
-    LOG_TOPIC("6507f", FATAL, Logger::STARTUP) << "Plan/" << agencyListKey << " in agency is no object. "
-                                      << "Agency not initialized?";
+    LOG_TOPIC("6507f", FATAL, Logger::STARTUP)
+        << "Plan/" << agencyListKey << " in agency is no object. "
+        << "Agency not initialized?";
+    return false;
+  }
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/// @brief create an id for a specified role
+//////////////////////////////////////////////////////////////////////////////
+
+bool ServerState::registerAtAgencyPhase1(AgencyComm& comm, ServerState::RoleEnum const& role) {
+  std::string const agencyListKey = roleToAgencyListKey(role);
+  std::string const latestIdKey = "Latest" + roleToAgencyKey(role) + "Id";
+
+  VPackBuilder builder;
+  builder.add(VPackValue("none"));
+
+  if(!checkIfAgencyInitialized(comm, role)) {
     return false;
   }
```

Comment:
You could create the builder after the check, would make the early-exit a little easier to spot.